src/ucp/core/ucp_context.h

```diff
@@ -643,10 +645,45 @@ int ucp_is_scalable_transport(ucp_context_h context, size_t max_num_eps)
     return (max_num_eps >= (size_t)context->config.est_num_eps);
 }
 
+static UCS_F_ALWAYS_INLINE int
+ucp_context_usage_tracker_enabled(ucp_context_h context)
+{
+    return context->config.ext.dynamic_tl_switch_interval != UCS_TIME_INFINITY;
+}
+
+static UCS_F_ALWAYS_INLINE double
+ucp_context_get_est_num_eps(ucp_context_h context,
+                            const ucs_linear_func_t *latency,
+                            int is_prioritized)
+{
+    if (!ucp_context_usage_tracker_enabled(context)) {
+        return context->config.est_num_eps;
+    }
+
+    if ((latency->m > 0) && !is_prioritized) {
+        return INFINITY;
+    }
+
+    return is_prioritized ? ucs_min(context->config.est_num_eps,
+                                    context->config.ext.max_priority_eps) :
+                            context->config.est_num_eps;
+}
+
+static UCS_F_ALWAYS_INLINE double
+ucp_tl_iface_latency_with_priority(ucp_context_h context,
+                                   const ucs_linear_func_t *latency,
+                                   int is_prioritized)
+{
+    unsigned num_eps = ucp_context_get_est_num_eps(context, latency,
```

Comment:
we could inline the code of ucp_context_get_est_num_eps function inside ucp_tl_iface_latency_with_priority. the only place where it's used separately - ucp_wireup_tl_iface_latency - could create a local linear function struct to pass to ucp_tl_iface_latency_with_priority.
this will simplify the code and avoid multiplying latency->m by INFINITY